### DISCUSSION TITLE
Error when using Team Red Miner on Ergo Pool

### DISCUSSION THREAD

**Discussion by MiningCryptoLive** [coinfoundry/miningcore#1217] [Q&A]
I get this error when trying to use Team Red Miner on  my Ergo Pool. Any help would be appreciated.

![Screenshot 2022-03-29 070403](https://user-images.githubusercontent.com/1049926/160598232-7f0f5f67-a6c9-4b89-8e4f-d553393e4c2e.png)

**Comment by hznovich** [comment]
Did you find a solution to the problem?

**Reply by MiningCryptoLive** [reply]
Not yet.

**Reply by MiningCryptoLive** [reply]
Updated to the latest version of TRM and it works now. Bypasses the rpc error

**Reply by oliverw** [reply]
Interesting.

**Reply by hznovich** [reply]
Does not work. Tried different settings in TRM, nothing helps. Post your config please, if it's not difficult...

**Reply by MiningCryptoLive** [reply]
> Does not work. Tried different settings in TRM, nothing helps. Post your config please, if it's not difficult...

teamredminer.exe -a autolykos2 --pool_broken_rpc -o stratum+ssl://erg.miningcrypto.live:3194 -u 9h9GhztZrz3HV9zwNhXcJJJo3TCcfLUC9Z48cJqcQ2DMArzdAUU.MCL -p x

**Reply by furcalor** [reply]
Which version on miningcore are you using? Seems to work on your pool, but not on others.

**Reply by MiningCryptoLive** [reply]
> Which version on miningcore are you using? Seems to work on your pool, but not on others.

I am using the latest release v69

**Reply by furcalor** [reply]
Well your lucky, fresh pull and build of v69 and still getting the same error myself.

**Reply by MiningCryptoLive** [reply]
> Well your lucky, fresh pull and build of v69 and still getting the same error myself.

That's weird. Maybe something different in the config file. Can you post yours?
also, send me your pool info and I will try to connect and see if I get the same thing.

**Reply by furcalor** [reply]
`{
    "logging":{
       "level": "trace",
       "enableConsoleLog": true,
       "enableConsoleColors": true,
       "logFile": "ergodebug.log",
       "apiLogFile": "ergoapi.log",
       "logBaseDirectory": "/home/ubuntu/testpoollogs",
       "perPoolLogFile": true
    },
    "api": {
        "enabled": true,
        "listenAddress": "0.0.0.0",
         "sslConfig": {
           "enabled": false,
           "sslPath": "",
           "sslPassword": "gjfgifdjgidf"
        },
        "port": 4011,
        "rateLimiting": {
            "disabled": true,
            "rules": [
                {
                    "Endpoint": "*",
                    "Period": "1s",
                    "Limit": 20
                }
            ],
            "ipWhitelist": [
                ""
            ]
        }
    },
    "banning":{
       "banOnJunkReceive": false,
       "banOnInvalidShares": false
    },
    "notifications": {
        "enabled": false,
        "email": {
            "host": "",
            "port": 587,
            "user": "",
            "password": "",
            "fromAddress": "",
            "fromName": ""
        },
        "admin": {
            "enabled": false,
            "emailAddress": "",
            "notifyBlockFound": true
        }
    },
    "persistence": {
        "postgres": {
            "host": "127.0.0.1",
            "port": 5432,
            "user": "postgres",
            "password": "removed",
            "database": "miningcore"
        }
    },
    "paymentProcessing":{
       "enabled": true,
       "interval": 600,
       "shareRecoveryFile": "recovered-shares.txt"
    },
    "pools": [
        {
                "id": "ergopplns",
                "enabled": true,
                "coin": "ergo",
                "address": "9fAD5F8u6bemS5rquBB7iUouS1idjiUxW2WWdiUMjCfmu4xcoGU",
                "maxActiveJobs": 20,
                "blockRefreshInterval": 1000,
                "jobRebroadcastTimeout": 50,
                "clientConnectionTimeout": 600,
                "rewardRecipients": [
                {
                        "address": "9fAD5F8u6bemS5rquBB7iUouS1idjiUxW2WWdiUMjCfmu4xcoGU",
                        "percentage": 0.01
                }],
                "banning": {
                        "enabled": false,
                        "time": 180,
                        "invalidPercent": 90,
                        "checkThreshold": 150
                },
                "ports": {
                        "3366": {
                                "listenAddress": "0.0.0.0",
                                "difficulty": 0.1,
                                "name": "GPU Mining",
                                "varDiff": {
                                        "minDiff": 0.01,
                                        "targetTime": 15,
                                        "retargetTime": 90,
                                        "variancePercent": 30,
                                        "maxDelta": 0.1
                                }
                        },
                        "13366": {
                                "listenAddress": "0.0.0.0",
                                "difficulty": 4000000000,
                                "name": "High Difficulty GPU Mining",
                                "varDiff": {
                                        "minDiff": 4000000000,
                                        "targetTime": 15,
                                        "retargetTime": 90,
                                        "variancePercent": 30,
                                        "maxDelta": 0.1
                                }
                        },
                        "23366": {
                                "listenAddress": "0.0.0.0",
                                "difficulty": 0.1,
                                "tls": true,
                                "tlsPfxFile": "/home/ubuntu/pool/cert.pfx",
                                "name": "SSL GPU Mining",
                                "varDiff": {
                                        "minDiff": 0.01,
                                        "targetTime": 15,
                                        "retargetTime": 90,
                                        "variancePercent": 30,
                                        "maxDelta": 0.1
                                }
                        }
                },
                "daemons": [
                {
                        "host": "127.0.0.1",
                        "port": 9053,
                        "apiKey": "removed"
                }
                ],
                "paymentProcessing": {
                        "enabled": true,
                        "walletPassword": "removed",
                        "minimumPayment": 0.01,
                        "payoutScheme": "PPLNS",
                        "payoutSchemeConfig": {
                                "factor": 1.0
                        }
                },
        },

]
}
`

**Reply by MiningCryptoLive** [reply]
Config looks fine to me. Whats the pool address and I will try to connect

**Reply by furcalor** [reply]
stratum+ssl://gb.mining4people.com:23194, but I doubt you have any different result there.
when connecting to yours, it seems like the miner gets subsribed before the pools responds with the RPC that TRM rejects.

**Reply by MiningCryptoLive** [reply]
do you have the Tls password set in the config? I see you have TlsPfxFile you also need the password set for ssl to work.

"tlsPfxPassword": "password",

**Reply by MiningCryptoLive** [reply]
also the error on your pool is different than mine. yours has to do with get set difficulty. Is the port set up as static difficulty or a variable difficulty?

**Reply by furcalor** [reply]
VarDiff, its the same error you originally were getting. No password on the TLS file. SSL works fine on other coins. Just TRM doesn't work with ERGO or any other coin.

**Reply by MiningCryptoLive** [reply]
I don't see how ssl is working without the password set. When I connect to your ergo pool I get a totally different error than when I connect to mine. My pool error is failed to parse rpc
When connecting to your pool it is a miner subscribe set difficulty error.

**Reply by furcalor** [reply]
Yeah its error you were originally getting in the picture you started the discussion with.
SSL does not have a password set, since the SSL file used for the stratum was generated without a password.

**Reply by furcalor** [reply]
`[2022-04-25 16:12:31] [Pool gb.mining4people.com] Send:
{"id":1,"method":"mining.subscribe","params":["teamredminer/0.9.4.2"]}

[2022-04-25 16:12:31] Dev pool connected and ready.
[2022-04-25 16:12:31] [Pool gb.mining4people.com] Recv:
{"result":[[["mining.set_difficulty","0HMH55IIEBEAG"],["mining.notify","0HMH55IIEBEAG"]],"400c",6],"id":1}
[2022-04-25 16:12:31] Pool gb.mining4people.com failed to parse server rpc: {"result":[[["mining.set_difficulty","0HMH55IIEBEAG"],["mining.notify","0HMH55IIEBEAG"]],"400c",6],"id":1}`

**Reply by MiningCryptoLive** [reply]
In TRM config did you set --pool_broken_rpc

**Reply by furcalor** [reply]
Well gonna keep tinkering with this, If I find a way to make it work will submit a pull requests. Thank you for taking the time to try to troubleshoot :)

**Reply by furcalor** [reply]
Yes, like said works fine with your pool and fails when I change it to our pool, or any other miningcore pool that has ergo. But at the same time works just fine with nvidia miners.

**Reply by MiningCryptoLive** [reply]
no problem at all. I will keep tinkering also. There has to be a fix

**Reply by goodthebest** [reply]
@furcalor did you figure out a solution yet?

**Reply by Censseo** [reply]
I am facing the same error, did u find a solution?
And it is not on SSL, but on regular connection (stratum+tcp)

**Comment by MiningCryptoLive** [comment]
![Screenshot 2022-04-24 071137](https://user-images.githubusercontent.com/1049926/164973670-dbc60b65-218d-4c2c-a840-9338b9151ba2.png)

**Comment by MiningCryptoLive** [comment]
Gotcha on the ssl.

**Comment by Censseo** [comment]
FYI it will be fix in the next release of TRM.

**Reply by MiningCryptoLive** [reply]
Awesome 👍
